core/common/src/test/java/alluxio/CommonTestUtils.java

```diff
@@ -54,4 +78,62 @@
     T finalObject = (T) current;
     return finalObject;
   }
+
+  /**
+   * Uses reflection to test the equals and hashCode methods for the given simple java object.
+   *
+   * It is required that the given class has a no-arg constructor.
+   *
+   * Note: To use this method to test a class which contains a final class as a field, you must
+   * prepare the final class for testing. See the top of {@link CommonTestUtilsTest} for an example.
+   *
+   * @param clazz the class to test the equals and hashCode methods for
+   * @param excludedFields names of fields which should not impact equality
+   */
+  public static <T> void testEquals(Class<T> clazz, String... excludedFields) throws Exception {
+    Set<String> excludedFieldsSet = Sets.newHashSet(Arrays.asList(excludedFields));
```

Comment:
I wonder whats the limitation of this test. Would it support the fields of references that lead to a cyclic graph?